File: 2021/07/22/main.py
class Solution:
    def partitionDisjoint(self, nums: List[int]) -> int:
        p, n, mx = 0, nums[0], nums[0]
        for i in range(len(nums)):
            mx = max(mx, nums[i])
            if nums[i] < n:
                p = i
                n = mx
        return p + 1

File: 2021/07/22/main.go
package main

func partitionDisjoint(nums []int) int {
	p, n, mx := 0, nums[0], nums[0]
	for i := 0; i < len(nums); i++ {
		if mx < nums[i] {
			mx = nums[i]
		}
		if nums[i] < n {
			p = i
			n = mx
		}
	}
	return p + 1
}
